subiquity/common/filesystem/tests/test_gaps.py

```diff
@@ -31,26 +32,74 @@
 
 class TestGaps(unittest.TestCase):
     def test_basic(self):
-        model, disk1 = make_model_and_disk()
-
-        pg = gaps.parts_and_gaps(disk1)
-        self.assertEqual(1, len(pg))
-        self.assertTrue(isinstance(pg[0], gaps.Gap))
-        self.assertEqual(MiB, pg[0].offset)
+        [gap] = gaps.parts_and_gaps(make_disk())
+        self.assertTrue(isinstance(gap, gaps.Gap))
+        self.assertEqual(MiB, gap.offset)
+
+    def test_copy(self):
+        [gap] = gaps.parts_and_gaps(make_disk())
+        actual = gap.copy()
+        self.assertEqual(gap, actual)
+        self.assertIsNot(gap, actual)
+
+
+class TestSplitGap(unittest.TestCase):
+    def test_equal(self):
+        [orig_gap] = gaps.parts_and_gaps(make_disk())
+        gap = orig_gap.copy()
+        actual = gap.split(gap.size)
+        self.assertEqual((gap, None), actual)
+
+    def test_too_big(self):
+        [orig_gap] = gaps.parts_and_gaps(make_disk())
+        gap = orig_gap.copy()
+        with self.assertRaises(Exception):
+            gap.split(gap.size + MiB)
+        self.assertEqual(orig_gap, gap)
+
+    def test_split(self):
+        [gap] = gaps.parts_and_gaps(make_disk(size=100 << 30))
+        size = 10 << 30
+        new_gaps = gap.split(size)
+        self.assertEqual(2, len(new_gaps))
+        self.assertEqual(size, new_gaps[0].size)
+        self.assertEqual(gap.size - size, new_gaps[1].size)
+        self.assertEqual(gap.offset, new_gaps[0].offset)
+        self.assertEqual(gap.offset + size, new_gaps[1].offset)
+
+
+class TestAtOffset(unittest.TestCase):
+    def test_zero(self):
+        self.assertIsNone(gaps.at_offset(make_disk(), 0))
+
+    def test_one_mib(self):
+        self.assertIsNotNone(gaps.at_offset(make_disk(), MiB))
```

Comment:
The MiB is kind of a mystery guest here. d=make_disk(); g=largest_gap(d); assertEqual(gaps.at_offset(d, g.offset), g) seems like a better test.

Reply:
Fixed, thanks.
I left the zero one, there will always be partition junk there.